### PR TITLE
[0.24] KafkaChannel dispatcher offset checking improvements

### PR DESCRIPTION
**Pull request by aliok**
Fixes #

## Proposed Changes

- Fixing some issues created during https://github.com/knative-sandbox/eventing-kafka/pull/886
- Applying the fixes from https://github.com/aliok/eventing-kafka/pull/1 (that was on main)
- Change offset checking loop
- Requeue channel when there's a failure in the loop

**Release Note**

```release-note
- 🐛  Fixed an issue with the offset checking in the dispatcher so that the reconciliation is more proper
```

**Docs**

### CODE REVIEW

**Comment by pierDipi** [comment]
@googlebot I consent.

**Comment by aliok** [comment]
/test pull-knative-sandbox-eventing-kafka-unit-tests

```
=== RUN   TestEnableSaramaLogging
==================
WARNING: DATA RACE
```

I think this test is disabled in later branches

**Comment by aliok** [comment]
/test pull-knative-sandbox-eventing-kafka-upgrade-tests

It is stuck

File: pkg/channel/consolidated/dispatcher/dispatcher.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ type KafkaDispatcher struct {
 	logger    *zap.SugaredLogger
 }
 
-func NewDispatcher(ctx context.Context, args *KafkaDispatcherArgs) (*KafkaDispatcher, error) {
+func NewDispatcher(ctx context.Context, args *KafkaDispatcherArgs, enqueue func(ref types.NamespacedName)) (*KafkaDispatcher, error) {
```
Comment (matzew) [inline pkg/channel/consolidated/dispatcher/dispatcher.go:76]: nit: lets add docs... ? at least on the PR for `main`

File: pkg/common/consumer/consumergroup_offsets_checker.go [hunk 2]
```diff
@@ -63,10 +64,7 @@ func (k *KafkaConsumerGroupOffsetsChecker) WaitForOffsetsInitialization(ctx cont
 			return false, fmt.Errorf("error checking if offsets are initialized. stopping trying. %w", err)
 		}
 	}
-	pollCtx, pollCtxCancel := context.WithTimeout(ctx, OffsetCheckRetryTimeout)
-	err = wait.PollUntil(OffsetCheckRetryInterval, check, pollCtx.Done())
-	defer pollCtxCancel()
-
+	err = wait.PollImmediate(OffsetCheckRetryInterval, OffsetCheckRetryTimeout, check)
```
Comment (matzew) [inline pkg/common/consumer/consumergroup_offsets_checker.go:67]: NIT: `check` could we name the conditional function to something more meaningful ?

like `checkIfAllOffsetsInitialized` ? 

(at least on the PR to `main` branch)

Comment (aliok) [reply pkg/common/consumer/consumergroup_offsets_checker.go:67]: +1. can do in main

Comment (aliok) [reply pkg/common/consumer/consumergroup_offsets_checker.go:67]: Addressed in main: https://github.com/knative-sandbox/eventing-kafka/pull/911

**Pull request merged**